pkg/controller/volume/attachdetach/metrics/metrics.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+var (
+	inUsePVCMetricDesc = prometheus.NewDesc(
+		prometheus.BuildFQName("", "storage_counter", "pvc_in_use"),
+		"Measure number of pvcs in use",
+		[]string{"node", "volume_plugin"}, nil)
+)
+var registerMetrics sync.Once
+
+type pvcInUseCollector struct {
+	pvcLister       corelisters.PersistentVolumeClaimLister
+	podLister       corelisters.PodLister
+	pvLister        corelisters.PersistentVolumeLister
+	volumePluginMgr *volume.VolumePluginMgr
+}
+
+type nodePVCCount map[types.NodeName]map[string]int
```

Comment:
Can you add a comment here describe what each of the types represents

Reply:
fixed